src/gwt/src/org/rstudio/studio/client/common/satellite/Satellite.java

```diff
@@ -191,6 +191,10 @@ private native void initializeNative(String name) /*-{
                $wnd.opener.notifyRStudioSatelliteClosed(name);
             }),
             true);
+      
+      // inherit desktop frame from opener
+      $wnd.desktop = $wnd.opener.desktop;
+      $wnd.desktopMenuCallback = $wnd.opener.desktopMenuCallback;
```

Comment:
Elsewhere we handle this by talking to the objects on the other window directly (instead of copying them to the current window object). For example:

https://github.com/rstudio/rstudio/blob/7bbb81720692a19eef58599dc42ceea0a415c9e5/src/gwt/src/org/rstudio/core/client/theme/ThemeFonts.java#L44-L59

 Either way is fine but it might be confusing to have both. Could we either (a) put this behind an accessor that delegates to the right object as done above, or (b) copy `desktopInfo` too so we don't need accessors?

Reply:
`DesktopFrame.java`, whose API maps to `window.desktop`, itself is just an interface (it actually gets replaced on startup with the QWebChannel object created by Qt) so it's a somewhat awkward place to attempt pushing a potential accessor. I think it'll be simplest to just expose `desktopInfo` on the satellite as I'm doing in this PR -- I'll make that change.

Reply:
Sounds great, thanks!

Reply:
(For reference, I think previously these objects were automatically propagated to satellite windows when using WebKit; this is no longer the case and so we need to manually make them accessible when the satellite is constructed)